over-tailing leads to batches with nil records

With a1e575a0a978a0019aef2e37faa4a1098a95ce1f: 

```
$ cat every.tzng 
#0:record[key1:string,key2:string,n:int32]
0:[a;x;1;]
0:[a;y;2;]
0:[b;z;1;]
#1:record[key1:ip,n:int32]
1:[10.0.0.1;1;]
1:[10.0.0.2;1;]
1:[10.0.0.1;1;]
11:45 ~/work/brim/zq(master)
$ zq -t "tail 100  " every.tzng 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x131e247]

goroutine 1 [running]:
github.com/brimsec/zq/zio/tzngio.(*Writer).Write(0xc00000fee0, 0x0, 0x0, 0x0)
        /Users/henridf/work/brim/zq/zio/tzngio/writer.go:38 +0x37
github.com/brimsec/zq/driver.(*CLI).Write(0xc0002b0570, 0x0, 0x1a4e460, 0xc0002c43c0, 0x0, 0x0)
        /Users/henridf/work/brim/zq/driver/driver.go:75 +0xcd
github.com/brimsec/zq/driver.Run(0xc0000cea00, 0x1a4c2e0, 0xc0002b0570, 0x0, 0xc00000ff00, 0xc0002bcff0)
        /Users/henridf/work/brim/zq/driver/driver.go:46 +0x248
main.(*Command).Run(0xc0001a48f0, 0xc000032120, 0x2, 0x2, 0x0, 0x0)
        /Users/henridf/work/brim/zq/cmd/zq/zq.go:263 +0x87a
github.com/mccanne/charm.(*instance).run(0xc00000fd80, 0xc000032110, 0x3, 0x3, 0x0, 0x0)
        /Users/henridf/go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:53 +0x28c
github.com/mccanne/charm.(*Spec).ExecRoot(0x1f25ec0, 0xc000032110, 0x3, 0x3, 0xc00029ff50, 0x100736f, 0xc0000a2058, 0x0)
        /Users/henridf/go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:77 +0x96
main.main()
        /Users/henridf/work/brim/zq/cmd/zq/main.go:9 +0x78

```

--- a/proc/tail.go
+++ b/proc/tail.go
@@ -22,9 +22,13 @@ func (t *Tail) tail() zbuf.Batch {
 	if t.count <= 0 {
 		return nil
 	}
-	out := make([]*zng.Record, t.limit)
-	for k := 0; k < t.limit; k++ {
-		out[k] = t.q[(t.off+k)%t.limit]
+	start := t.off
+	if t.count < t.limit {
+		start = 0
+	}
+	out := make([]*zng.Record, t.count)
+	for k := 0; k < t.count; k++ {
+		out[k] = t.q[(start+k)%t.limit]
 	}
 	t.off = 0
 	t.count = 0

--- a/tests/suite/tail/toomuch.yaml
+++ b/tests/suite/tail/toomuch.yaml
@@ -1,0 +1,9 @@
+zql: tail 2
+
+input: |
+  #0:record[x:string]
+  0:[first;]
+
+output: |
+  #0:record[x:string]
+  0:[first;]

